File: std/testdata/service/modeling/impl.go
// Copyright 2022 Namespace Labs Inc; All rights reserved.
// Licensed under the EARLY ACCESS SOFTWARE LICENSE AGREEMENT
// available at http://github.com/namespacelabs/foundation

package modeling

import (
	"context"

	"google.golang.org/protobuf/types/known/emptypb"
	"namespacelabs.dev/foundation/std/go/grpc/server"
	"namespacelabs.dev/foundation/std/testdata/scopes"
)

type Service struct {
	deps *ServiceDeps
}

func (svc *Service) GetScopedData(ctx context.Context, _ *emptypb.Empty) (*GetScopedDataResponse, error) {
	response := &GetScopedDataResponse{Item: []*scopes.ScopedData{
		svc.deps.One,
		svc.deps.Two,
	}}

	return response, nil
}

func WireService(ctx context.Context, srv *server.Grpc, deps *ServiceDeps) {
	svc := &Service{deps: deps}
	RegisterModelingServiceServer(srv, svc)
}

File: std/nodejs/monitoring/tracing/fastify/impl.ts
// Copyright 2022 Namespace Labs Inc; All rights reserved.
// Licensed under the EARLY ACCESS SOFTWARE LICENSE AGREEMENT
// available at http://github.com/namespacelabs/foundation

import openTelemetryPlugin from "@autotelic/fastify-opentelemetry";
import { ExtensionDeps } from "./deps.fn";

export const initialize = async (deps: ExtensionDeps) => {
	const server = await deps.httpServer;
	server.fastify().register(openTelemetryPlugin, { wrapRoutes: true });

	console.log(`Initialized OpenTelemetry instrumentation for the HTTP server.`);
};
